### PR TITLE
Check that nodes have a pre-assigned name

### PR DESCRIPTION
**Pull request by sidneymbell**
The current titer module implementation relies on pre-assigned node names to index the output `cTiter` and `dTiter` values. If passing in an externally-generated tree, nodes are not necessarily pre-named.

I added an assertion for now as a stop-gap. I think we have a few options that warrant discussion. 

Currently we export only a JSON file mapping node names to `cTiter` and `dTiter` values.

We have two decision points:
1 - Always require pre-assigned node names in the input phylogeny vs assign any missing node names
2 - Only export the JSON mapping node names to titer values vs also export a copy of the tree, annotated with names, `cTiter` and `dTiter` values. 

Note that if we do opt to assign missing node names, we should also export a tree with at least the assigned names annotated.

### CODE REVIEW

**Comment by tsibley** [comment]
I think that `augur titers` shouldn't mutate the tree, even just to add node names (as we talked about briefly in person). That is, I don't think it's consistent with other design decisions we've made if the `titers` commands output a new tree. This then supports the assertion that `augur titers` requires all nodes to be named. Alternatively, the command could only provide output for nodes that are named, ignoring ones which aren't.

Either way, we could facilitate node naming for folks bringing their own, non-augur-generated trees to the table by providing an ancillary command which takes a tree, assigns names to unnamed nodes, and outputs the new tree. This fits with the larger design decision of making small, composable commands that do a single thing.

BTW, though it might not matter for this branch, you probably want to take a look at the `seasonal_flu_mods` branch which has some additional proposed, work-in-progress changes to the `titers` command.

**Comment by emmahodcroft** [comment]
If internal node names are what are needed here (sorry, on phone a little
hard to follow well), this was discussed some time ago as being possible by
running `refine` without other arguments. I think this was implemented and
should work, but it's been a long time since I tried it.

**Comment by tsibley** [comment]
Ah! I vaguely recall that now. From a cursory glance at the code/comments it does seem like roughly `augur refine --tree foo.newick --output-tree foo_with_names.newick` would produce a tree with internal nodes named but nothing else of significance changed. I wonder if we should wrap up essentially that into a more transparent/obvious command (e.g. `augur name-all-nodes` or somesuch).

**Comment by emmahodcroft** [comment]
So the discussion I vaguely remember was that `refine` would be the one function that always should be run when using `augur` - whether to do time-resolution or just to get internal node names. This would then avoid confusion with people thinking they had to run both `refine` and something else, potentially re-naming internal nodes (with different names) part-way through analysis. 

Used with snakemake, this should also guard against this happening - `refine` comes before any of the more complex analyses, and so if it's re-run, all downstream steps will re-run as well. But I do think we can make this much clearer, via docs and a tutorial for 'what if I already have a tree'.

**Comment by rneher** [comment]
yes, part of the reason `refine` has this bizarre name is that it serves (too) many purposes including just renaming nodes, time tree estimation, rerooting etc. But it basically serves as an entry point for trees into the augur pipeline. still, an assertion that all nodes are named makes sense to guarantee sanity of later steps.

**Comment by jameshadfield** [comment]
Agreed about this -- @evogytis and I struggled with BEAST trees regarding internal nodes etc and ended up copying a few lines of code from `refine.py` -- https://github.com/nextstrain/augur/blob/beast/augur/import_beast.py#L363

**Comment by tsibley** [comment]
This all sounds reasonable to me! I agree with Emma that we can make it much clearer with docs and tutorial and with Richard that the assertion is still useful to ensure sanity.

**Comment by sidneymbell** [comment]
Awesome, thanks everyone. I'll update the `Exception` stdout to point users to `refine`. OK to merge at that point?

**Comment by rneher** [comment]
Would be easier if this was just on top of `seasonal_flu_mods`. In terms of titer model development, this will be the branch going forward. Merging this one line now into master will only generate a conflict. My proposal: just commit this one line to 'seasonal_flu_mods` and delete this branch...

**Comment by sidneymbell** [comment]
Cool, will do. PR was just to make a spot to have the discussion re: naming nodes. Closing now. :)
